File: lab_3/30/lab30.py
def main():
    while True:
        choice = int(input("โปรดเลือกเมนู:\n1. [ For loop ]\n2. [ While loop ]\n3. [ Do-while loop ]\nเลือก: "))

        match choice:
            case 1:
                for i in range(1, 11):
                    print(f"{i:3}", end="")
            case 2:
                i = 1
                while i <= 10:
                    print(f"{i:3}", end="")
                    i += 1
            case 3:
                i = 1
                while True:
                    print(f"{i:3}", end="")
                    i += 1
                    if i > 10:
                        break
            case _:
                print("\nเมนูไม่ถูกต้อง!")

        choice = int(input("\nกด 0 เพื่อออกจากโปรแกรมหรืออื่นๆเพื่อใช้งานต่อ: "))
        if choice == 0:
            break

if __name__ == "__main__":
    main()

File: lab_3/30/lab30.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int main() {
    while(true){
    int choice;
    cout << "โปรดเลือกเมนู:\n";
    cout << "1. [ For loop ]\n";
    cout << "2. [ While loop ]\n";
    cout << "3. [ Do-while loop ]\n";
    cout << "เลือก: ";
    cin >> choice;

    switch (choice) {
        case 1:
            for (int i = 1; i <= 10; i++) {
                cout << i << setw(3);
            }
            break;
        case 2:
            {
                int i = 1;
                while (i <= 10) {
                    cout << i << setw(3);
                    i++;
                }
            }
            break;
        case 3:
            {
                int i = 1;
                do {
                    cout << i << setw(3);
                    i++;
                } while (i <= 10);
            }
            break;
        default:
            cout << endl << "เมนูไม่ถูกต้อง!" << endl;
            break;
    }

    cout << endl << "กด 0 เพื่อออกจากโปรแกรมหรืออื่นๆเพื่อใช้งานต่อ: ";
    cin >> choice;
    if (choice == 0) {
        break;
        }
    }

    return 0;
}
